File: backend/src/api/routes/internal_notes.py
"""Internal note API routes (internal team only)."""

from fastapi import APIRouter, Depends, status

from src.api.dependencies import DBSession, get_current_user, get_project_for_internal
from src.models.project import Project
from src.models.user import User
from src.schemas.internal_note import (
    InternalNoteCreate,
    InternalNoteReplyCreate,
)
from src.services.internal_note import create_note, create_reply, list_notes

router = APIRouter(prefix="/projects/{project_id}/internal-notes", tags=["Internal Notes"])


@router.get("")
async def get_notes(
    project_id: int,
    db: DBSession,
    project: Project = Depends(get_project_for_internal),
):
    return {"notes": await list_notes(db, project_id)}


@router.post("", status_code=status.HTTP_201_CREATED)
async def post_note(
    project_id: int,
    data: InternalNoteCreate,
    db: DBSession,
    current_user: User = Depends(get_current_user),
    project: Project = Depends(get_project_for_internal),
):
    return await create_note(db, project, current_user, data.body, data.mentions)


@router.post("/{note_id}/replies", status_code=status.HTTP_201_CREATED)
async def post_reply(
    project_id: int,
    note_id: int,
    data: InternalNoteReplyCreate,
    db: DBSession,
    current_user: User = Depends(get_current_user),
    project: Project = Depends(get_project_for_internal),
):
    return await create_reply(db, project_id, note_id, current_user.id, data.body)

File: frontend/src/api/endpoints/internalNotes.ts
import apiClient from '../client';
import type { InternalNote, InternalNotesResponse } from '../../types';

export const listInternalNotes = (projectId: number) =>
  apiClient.get<InternalNotesResponse>(`/projects/${projectId}/internal-notes`).then((r) => r.data);

export const createInternalNote = (projectId: number, data: { body: string; mentions: number[] }) =>
  apiClient.post<InternalNote>(`/projects/${projectId}/internal-notes`, data).then((r) => r.data);

export const createInternalNoteReply = (projectId: number, noteId: number, body: string) =>
  apiClient
    .post(`/projects/${projectId}/internal-notes/${noteId}/replies`, { body })
    .then((r) => r.data);
